File: test-dotnet/test-dotnet/Controllers/ProductController.cs
using Microsoft.AspNetCore.Mvc;
using Npgsql;
using System;
using System.Collections;
using System.Threading;
using System.Threading.Tasks;
using TestPerf.Model;

namespace TestPerf.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class ProductController : ControllerBase
    {

        [HttpPost]
        [Route("createProduct")]
        public async Task CreateProduct(ProductModel product, CancellationToken cancellationToken) 
        {
            var connString = "Host={HOST};Username={USERNAME};Password={PASSWORD};Database={DATABASE};SSL Mode=Require;Trust Server Certificate=true";

            await using var conn = new NpgsqlConnection(connString);
            await conn.OpenAsync(cancellationToken);

            await using (var cmd = new NpgsqlCommand("INSERT INTO test.perftest (\"name\", price, category, description)VALUES(@pname, @pprice, @pcategory, @pdesc);", conn))
            {
                cmd.Parameters.AddWithValue("pname", product.Name);
                cmd.Parameters.AddWithValue("pprice", product.Price);
                cmd.Parameters.AddWithValue("pcategory", product.Category);
                cmd.Parameters.AddWithValue("pdesc", product.Description);

                await cmd.ExecuteNonQueryAsync(cancellationToken);
            }

            await conn.CloseAsync();
        }

        [HttpGet]
        [Route("ping")]
        public Task<string> Ping()
        {
            return Task.FromResult("OK");
        }

        [HttpGet]
        [Route("listVars")]
        public IDictionary ListVars() 
        {
            var vars = Environment.GetEnvironmentVariables();

            return vars;
        }
    }
}

File: test-node/controllers/product.controller.js
const Product = require('../models/product.model');
const db = require('../pool');

exports.listVars = async function listVars (req, res, next) {
    return res.json(process.env);
}

exports.ping = async function ping (req, res, next) {
    return res.json("OK");
}

exports.createProduct = async function createProduct (req, res, next) {
    
    try{
        const client = await db.getPool().connect();

        try{
            const model = Product.fromReq({body:req.body});
        
            await client.query("INSERT INTO test.perftest (\"name\", price, category, description)VALUES($1, $2, $3, $4);", [
                model.name,
                model.price,
                model.category,
                model.description
            ]);
    
        } finally{
            client.release();
        }
    
        res.end();
    } catch {
        res.status(400).end();
    }
}